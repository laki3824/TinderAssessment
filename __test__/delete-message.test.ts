const getRecentMessageFromChannel = require( "./common-utils" );
const { WebClient } = require('@slack/web-api');
const dotenv = require('dotenv')

describe('Delete messages tests - ', () => {

    beforeAll(() => {
        console.log("Start tests....");
        dotenv.config();
    });
  
    afterAll(() => {
      console.log("End tests....")
    });

    /**
     * Delete a posted message from a channel.
     * https://api.slack.com/methods/chat.delete
     */
    test('Delete a text message using a valid messageId', async() => {
        //Post a message before trying to delete to keep this test independent of other tests
        const client = new WebClient(process.env.USER_API_KEY);

        const channelId = process.env.CHANNEL_ID;

        const response = await client.chat.postMessage({
            channel: process.env.CHANNEL_NAME,
            text: "This is a text message",
        });

        //Verify whether the message was posted to the channel
        expect(response.ok).toBe(true);

        //Retrieve the recent message from the channel
        const message = await getRecentMessageFromChannel(client, channelId);

        const messageId = message.ts;
        console.log("Retrieved messageId: " + messageId);
        
        //Edit the message
        const deleteResponse = await client.chat.delete({
            channel: channelId,
            ts: messageId
        });

        //Check whether the operation was successful.
        expect(deleteResponse.ok).toBe(true);

        //Verify whether the correct ts/messageId is deleted
        expect(deleteResponse.ts).toEqual(messageId);

        //Validate the response snap shot
        expect(deleteResponse).toMatchSnapshot({
            ts: expect.any(String)
        });

    });

});
