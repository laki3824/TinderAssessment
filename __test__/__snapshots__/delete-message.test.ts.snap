// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete messages tests -  Delete a text message using a valid messageId 1`] = `
Object {
  "channel": "C029Y3JNWTS",
  "ok": true,
  "response_metadata": Object {
    "acceptedScopes": Array [
      "chat:write",
    ],
    "scopes": Array [
      "identify",
      "channels:history",
      "chat:write",
    ],
  },
  "ts": Any<String>,
}
`;
